synapse/rest/client/login.py

```diff
@@ -166,12 +166,10 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, LoginResponse]:
         if self._msc2918_enabled:
             # Check if this login should also issue a refresh token, as per MSC2918
             should_issue_refresh_token = login_submission.get(
-                "org.matrix.msc2918.refresh_token", False
+                LoginRestServlet.REFRESH_TOKEN_PARAM, False
```

Comment:
Any reason for spelling the class name explicitly here, rather than using `self`?

Reply:
(fwiw the rest of this class does that for all the constants)

This is probably a Javaism, where accessing static fields/methods using `this` (`self`) is discouraged as it camouflages the fact that it's static.
I don't know if it's also bad practice in Python code and no linters told me off for it, but it's something I still think I'd avoid for clarity.

Maybe I'm just a fool?

Reply:
Don't have an authoritative answer for you other than "it looks weird to me". Not worth worrying about it though.